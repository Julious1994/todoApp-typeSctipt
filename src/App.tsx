import * as React from 'react';
import './App.css';
import { HandleOnChangeInterface } from './commonInterface';
import Form from './Form';
import List from './list';


interface props {};
interface state {
  text: string,
  list: Array<any>
};

class App extends React.Component<props, state> {

  state: Readonly<state> = {
    text: '',
    list: [],
  };

  onChange = (e: HandleOnChangeInterface) => {
    this.setState({ text: e.target.value});
  }

  saveTodo = () => {
    const { list, text } = this.state;
    if(text !== '') {
      list.push({ text, selected: false });
      this.setState({ text: '', list: [...list] });
    }
  }

  selectTodo = (index: number) => {
    const { list } = this.state;
    const todo = list[index];
    todo.selected = !todo.selected;
    list[index] = {...todo};
    this.setState({ list });
  }

  render() {
    var { text, list } = this.state;
    return (
      <div className="App">
        <h1>Todo app</h1>
        <Form 
          text={text}
          onChange={this.onChange}
          onClick={this.saveTodo}
        />
        <List 
          list={list}
          onClick={this.selectTodo}
        />
      </div>
    );
  }
}

export default App;
