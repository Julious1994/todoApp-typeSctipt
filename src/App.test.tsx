import * as React from 'react';
import { shallow } from 'enzyme';
import App from './App';

let wrapper: any;
beforeEach(() => {
  wrapper = shallow(<App />);
})

it('renders without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});
it('should render Todo App', () => {
  const headerText = wrapper.find("h1").at(0).text();
  expect(headerText).toBe("Todo app");
});

it('should change text input', () => {
  const form = wrapper.find('Form').at(0);
  form.props().onChange({ target: { value: 'Test'}});
  expect(wrapper.state().text).toBe('Test');
});

it('should save todo', () => {
  const form = wrapper.find('Form').at(0);
  wrapper.setState({ text: "Test"});
  form.props().onClick();
  expect(wrapper.state().list.length).toBe(1);
});

it('should select todo', () => {
  const list = [{ text: "Test", selected: false },{ text: "Test1", selected: false }];
  wrapper.setState({ list });
  wrapper.instance().selectTodo(1);
  const stateList = wrapper.state().list;
  expect(stateList[1].selected).toBeTruthy();
});

it('should not save todo', () => {
  const form = wrapper.find('Form').at(0);
  form.props().onClick();
  expect(wrapper.state().list.length).toBe(0);
});
