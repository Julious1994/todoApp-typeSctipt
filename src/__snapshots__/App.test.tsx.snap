// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Todo app
        </h1>,
        <Form
          onChange={[Function]}
          onClick={[Function]}
          text=""
        />,
        <List
          list={Array []}
          onClick={[Function]}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Todo app",
        },
        "ref": null,
        "rendered": "Todo app",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
          "onClick": [Function],
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "list": Array [],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Todo app
          </h1>,
          <Form
            onChange={[Function]}
            onClick={[Function]}
            text=""
          />,
          <List
            list={Array []}
            onClick={[Function]}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Todo app",
          },
          "ref": null,
          "rendered": "Todo app",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
            "onClick": [Function],
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "list": Array [],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
